main.cpp

```diff
@@ -11,16 +11,8 @@ int main (int argc, char* argv[]) {
 	std::cerr << "Invalid call. Please call ./webserver <config_file>" <<std::endl;
 	return -1;
 	}
-
-  	int port;
-	
-	ServerConfig config(argv[1]);
-	
-	port = config.GetPortNo();
-	std::cout << port;
 	
-	http::server::server server_("127.0.0.1", std::to_string(port), &config);
-	//server_.config = &config;
+	http::server::server server_("127.0.0.1", argv[1]);
```

Comment:
I really like how you removed the code of trying to get the port number and passing that in to the server constructor. Now everything is done internally.